docs/app-design.md

```diff
@@ -0,0 +1,148 @@
+* Errata can have various resolutions
+  * Duplicate
+  * Not An Error
+  * Will Not Fix
+  * Approved
+  * Major Contains code for the BlogBook Revision
+  * Technical Error
+  * Sent to Customer Support
+  * More Information Requested
+* Errata can be various error types
+  * Other factual inaccuracy in content
+  * General/pedagogical suggestion or question
+  * Incorrect answer, calculation, or solution
+  * Broken link
+  * Typo
+  * Other
+* Emails are sent to users who entered the errata as it moves through the workflow. users can opt out of receiving the emails.
+* Errata can be filtered in the UI
+
+### Extra Admin Filters
+* Used to filter errata
+* Allows users to filter multiple books in the same query
+
+### Global Settings
+* Views and models for items on the Admin Settings menu
+  * Sticky Note
+  * Footer
+  * Cloudfront Distribution 
+
+### Mail
+* Mainly used to send the redirect report email to the Content team
+* There could be some dead code in this app
+
+### News
+* Contains code for the Blog Atom feed
+* Contains code for blog posts
+* Has APIs for a full list of blog posts or an single blog post
+* Contains code for OS press kit
+* Search is implemented for blog posts
+
+### OpenStax
+* The application used for settings and general items
+* APIs for accessing Documents and Images
+* Document API checks if faculty documents should be hidden or not
+* Routing for other APIs
+
+### OXAuth
+* Handles OAuth between CMS and OS Accounts
+* Has APIs for login, logout and user info. The user info API is not used in the OS app. It gets user info from the OS acccounts API
+
+### Pages
+* Models, APIs and views for all pages related to the OS site
+* Each page has a unique set of fields, images and/or documents.
+* All of the pages are children of the Homepage.
+* Has template for general pages that can be used for short term pages on the OS site
+
+### Redirects
+* Has management command that checks if all the redirect URLs return a 201. It creates a list of all failures and emails the list to the content team.
+* The command is run on the first of each month via a cron job
+
+### Salesforce
+* Used to pull data from Salesforce
+  * Adopter
+  * Faculty status
+  * Partners
+  * Schools and mapbox data
+  * Opportunities
+* Used to upload school data to mapbox
+* Uses SimpleSalesforce to connect to Salesforce API
+* Opportunities are used for the renewal form which users are asked to update after a period of time.
+
+### Snippets
+* Snippets are objects that can be reused on pages for specific content
+* They can be created, edited or removed in the Admin interface
+* Current snippets are
+  * Subject
+  * Faculty Resourse
+  * Student Resource
+  * Role
+  * Shared Content
+  * News Source
+* There are APIs for Roles and Subjects
+
+## Other
+* **Import-Export:** Admin has a feature to import or export a page. This uses the Wagtail import-export package. It is used to move pages from one environment to another.
```

Comment:
Here is a link to this project: https://pypi.org/project/wagtail-import-export-tool/

Reply:
@mwharrison This has been added